server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/Keywords.scala

```diff
@@ -22,42 +22,42 @@ import javax.mail.Flags
 import org.apache.james.mailbox.FlagsBuilder
 
 import scala.collection.IterableOnce
+import scala.util.{Failure, Success, Try}
 
 trait ToKeyword {
   def toKeyword(value: String): Option[Keyword]
 }
 
+trait KeywordFilter extends IterableOnce[Keyword]
+
+trait KeywordsValidator {
+  def validate(keywords: Set[Keyword]): Try[Unit]
+}
+
 object ToKeyword {
-  val STRICT: ToKeyword = (value: String) => Some(Keyword.of(value))
+  val STRICT: ToKeyword = (value: String) => Some(Keyword.of(value).get)
```

Comment:
Maybe this ToKeyword stuff don't make sense anymore, and we should get rid of it, as the caller could instead decide what to do from the Either?